inginious/frontend/plugins/problem_bank/react_app/src/bank_page.jsx

```diff
@@ -0,0 +1,329 @@
+                    data: data,
+                    isVisibleAlert: true,
+                    titleAlert: "Success!",
+                    styleAlert: "success"
+                }
+            });
+        }).error((data) => {
+            this.setState( {
+                dataAlertCourseList: {
+                    isVisibleAlert: true,
+                    data: {"message": data["responseJSON"]["error"]},
+                    titleAlert: "Error!",
+                    styleAlert: "danger"
+                }
+            });
+        });
+    };
+
+    updateFilteredTasksAsync(query){
+        $.post( "/plugins/problems_bank/api/filter_bank_tasks", { "task_query": query }, (filteredTasks) => {
+            let newTotalPages = Math.ceil(filteredTasks.length / this.limit);
+            let newPage = this.state.pageTasks;
+            if( newTotalPages >= 1) {
+                if( this.state.pageTasks > newTotalPages){
+                    newPage = newTotalPages
+                }
+            } else {
+                newPage = 1;
+                newTotalPages = 1;
+            }
+
+            this.setState({
+                tasks : filteredTasks,
+                pageTasks : newPage,
+                totalPagesTasks : newTotalPages
+            });
+        });
+    }
+
+    addCourse(courseId){
+
+        $.post( "/plugins/problems_bank/api/bank_courses", { "course_id": courseId }, (data) => {
+            this.updateBankCoursesAsync();
+            this.updateAvailableCoursesAsync();
+            this.updateTasksAsync();
+        }).done((data) => {
+            this.setState( {
+                dataAlertCourseList:{
+                    data: data,
+                    isVisibleAlert: true,
+                    titleAlert: "Success!",
+                    styleAlert: "success"
+                }
+            });
+        }).error((data) => {
+            this.setState( {
+                dataAlertCourseList: {
+                    isVisibleAlert: true,
+                    data: {"message": data["responseJSON"]["error"]},
+                    titleAlert: "Error!",
+                    styleAlert: "danger"
+                }
+            });
+        });
+    }
+
+    onAlertTaskListClose(isVisible){
+        this.setState({
+           dataAlertTaskList: {
+               isVisibleAlert: isVisible,
+               data: {"message" : ""},
+               titleAlert: '',
+               styleAlert: ''
+           }
+        });
+    }
+
+    onAlertCourseListClose(isVisible){
+        this.setState({
+           dataAlertCourseList: {
+               isVisibleAlert: isVisible,
+               data: {"message" : ""},
+               titleAlert: '',
+               styleAlert: ''
+           }
+        });
+    }
+
+    addTaskToCourse(targetId, taskId, bankId){
+        $.post( "/plugins/problems_bank/api/copy_task",
+            {"target_id": targetId, "task_id": taskId, "bank_id": bankId} ,( data ) => {
```

Comment:
Are we using callbacks with promise resolve?

Reply:
We are using promise, as `.post` returns an object that implements the `Promise` interface giving it all the properties, methods, and behavior of a Promise. So we can handle when succeeds the request or there is an error. Also, the callback is called when the request succeeds.